models/team.py

```diff
@@ -0,0 +1,22 @@
+"""Module containing the Team class and other related components"""
+
+from __future__ import annotations
+from dataclasses import dataclass
+
+
+from match import Match
+
+@dataclass
+class Team:
+    """A football team playing in a particular season of the Premier League.
+
+    Instance Attributes:
+        - name: The name of this team.
+        - matches: A chronologically ordered list of the matches played by this team in the season.
```

Comment:
Noticed that you said a particular season. So would Chelsea have 8 classes for 8 seasons or just one class in total? Perhaps it would be good to have a `seasons` attribute for a list of seasons instead of a list of `matches`? We can then retrieve the match list from `seasons`.

Reply:
Yes, in this approach Chelsea would have 8 instances for 8 seasons. Since we are making graphs for each season, I think we should not use the same vertex in each graph. What do you think?

Reply:
I think its fine to use the same vertex in each graph since the team doesn't change only the games do. Also, if we ever do analysis over multiple seasons having to deal with 8 different Chelsea's would be a hassle.